src/control/types.ts

```diff
@@ -26,6 +26,13 @@ export const ReplicasSchema = positiveInteger;
 export const PodsSchema = positiveInteger;
 export const MetricSchema = nonemptyString;
 export const DimensionSchema = positiveInteger;
+export const RegionSchema = nonemptyString;
+export const CloudSchema = Type.Union([
+  Type.Literal('gcp'),
+  Type.Literal('aws'),
+  Type.Literal('azure'),
+]);
```

Comment:
There might be a better way of doing this, open to feedback. This also isn't handled gracefully by `errorFormatter` in `validator.ts`. I'd like to cut a ticket and update if using literals here is deemed the way to go.

Reply:
The specific changes to `validator.ts` to properly handle errors related to this schema are addressed here: https://github.com/pinecone-io/pinecone-ts-client/pull/154